### PR TITLE
chore: Add MSP to storage request

### PR DESCRIPTION
**Pull request by snowmead**

### CODE REVIEW

**Review by a-moreira — APPROVED** [review]
I'm not completely aware of all the requirements, but the changes look good to me 👍

**Review by ffarall — APPROVED** [review]

**Comment by ffarall** [comment]
Fix CI and then ready to merge. Good job!

File: pallets/file-system/src/lib.rs [hunk 1]
```diff
@@ -376,6 +378,7 @@ pub mod pallet {
             location: FileLocation<T>,
             fingerprint: Fingerprint<T>,
             size: StorageData<T>,
+            msp: ProviderFor<T>,
```
Comment (TDemeco) [inline pallets/file-system/src/lib.rs:381]: This might be more explicit with the `msp_id` naming

File: pallets/file-system/src/types.rs [hunk 2]
```diff
@@ -118,3 +121,6 @@ pub type MaxMultiAddresses<T> =
 
 /// Alias for a bounded vector of [`MultiAddress`].
 pub type MultiAddresses<T> = BoundedVec<MultiAddress<T>, MaxMultiAddresses<T>>;
+
+/// Alias for the `Provider` type used in the ProvidersInterface.
+pub type ProviderFor<T> = <<T as crate::Config>::Providers as ProvidersInterface>::Provider;
```
Comment (TDemeco) [inline pallets/file-system/src/types.rs:126]: Just as a comment, in my latest PR I have updated all `Provider` references in the interfaces to be called `ProviderId` to be more clear on what the type actually represents, that's why I'm preeliminary suggesting naming changes to keep up with that. I can make those changes in my PR if needed

File: pallets/file-system/src/types.rs [hunk 3]
```diff
@@ -24,6 +24,10 @@ pub struct StorageRequestMetadata<T: Config> {
     /// SPs will use this to determine if they have enough space to store the data.
     /// This is also used to verify that the data sent by the user matches the size specified here.
     pub size: StorageData<T>,
+    /// MSP who is requested to store the data.
+    ///
+    /// This is optional in the event when a storage request is created solely to replicate data to other BSPs and an MSP is already storing the data.
+    pub msp: Option<ProviderFor<T>>,
```
Comment (TDemeco) [inline pallets/file-system/src/types.rs:30]: Same here ^

File: pallets/file-system/src/utils.rs [hunk 4]
```diff
@@ -68,6 +68,7 @@ where
         location: FileLocation<T>,
         fingerprint: Fingerprint<T>,
         size: StorageData<T>,
+        msp: Option<ProviderFor<T>>,
```
Comment (TDemeco) [inline pallets/file-system/src/utils.rs:71]: Same here ^

**Comment by snowmead** [comment]
CI (Run Para & Relay Network Test) fixed in this https://github.com/Moonsong-Labs/storage-hub/pull/84

**Pull request merged**